### ISSUE TITLE
[Feature Request] Bookmark Assets API

### ISSUE THREAD

**Issue by tnt1232007** [sissbruecker/linkding#718]
### Scenario
Hi, I am thinking of using a LLM to summerize the articles that I saved in Linkding.

Currently I am trying to do it with n8n:
1. Fetch the bookmark article from Linkding
2. Run its content through a LLM to summerize
3. Save the response as an asset back to Linkding

But I find out there's no APIs available to read and upload assets afaik

### Requests
- Modify the existing GET Bookmark APIs to expose the Asset IDs
- New set of Bookmark Assets APIs:
  - GET - Retrieve asset metadata by ID, as showed in `/admin/bookmarks/bookmarkasset/<ASSET_ID>/change/`
  - GET - Retrieve asset data by ID, either raw content or file download
    Extra if I can utilize the new reader-mode and optionally retrieve the content that has been processed
  - POST - Upload new asset to a bookmark
  - DELETE - Delete asset by ID, just for completeness

Please review and consider the implemention.
Thanks for the amazing works, especially the recently added features, I love the snapshot and reader mode.

**Comment by sissbruecker** [comment]
Exposing a list of assets through the bookmark details, and then having an endpoint for downloading an asset by ID would probably be a good start. Contributing that shouldn't be too difficult if you're interested.

Regarding reader mode, converting the HTML into a readable article actually only happens in the browser when you open the reader view, so that content is not available on the server. Depending on which language you want to use for processing your bookmarks there is probably some similar library for extracting the readable content from an HTML document.

**Comment by tjofle** [comment]
It would also be nice to have an endpoint to **upload** of assets. Ideally so attachments could be uploaded during bookmark creation, similar to Hoarder.app`s support for SingleFile:
https://docs.hoarder.app/Guides/singlefile (REST API as Save Destination in SaveFile browser extension)

The use case is to run SingleFile in the browser, which works much better when websites have Captchas, login walls, cookie banners etc. Then save the link in LinkDing with the output from SingleFile as the snapshot.

**Comment by sissbruecker** [comment]
@tjofle I'd say that's a different feature, as you don't just want to upload an asset, but potentially create a new bookmark together with an asset instead. Could you please create a separate issue for that?

**Comment by tjofle** [comment]
You're right @sissbruecker , as that would be both creating a new bookmark and uploading an asset in the same call. 

I created a new Feature request: Add REST API support for being a compatible SingleFile Destination #980
